### PR TITLE
Refactor: SettingsManager + unit test

### PR DESCRIPTION
**Pull request by rhpijnacker**
I have refactored the retrieving and storing of the settings into a separate module and added unit tests for them.

### CODE REVIEW

File: v2g-liberty/rootfs/root/appdaemon/apps/__init__.py
Comment (rhpijnacker) [inline v2g-liberty/rootfs/root/appdaemon/apps/__init__.py:1]: This is needed to enable the tests to load the files in the apps folder as python modules.

File: v2g-liberty/.vscode/tasks.json [hunk 1]
```diff
@@ -9,11 +9,23 @@
       "command": "python -m appdaemon -c rootfs/root/appdaemon -C appdaemon.devcontainer.yaml -D INFO",
       "problemMatcher": []
     },
+    {
```
Comment (rhpijnacker) [inline v2g-liberty/.vscode/tasks.json:12]: Sometimes Ctrl-C does not seem to work, so I added an extra task to kill the running V2G-Librty app in the devcontainer.

File: v2g-liberty/pytest.ini [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[pytest]
+addopts = --ignore=rootfs/root/tests
```
Comment (rhpijnacker) [inline v2g-liberty/pytest.ini:2]: Ignoring the time_slot tests for now, until they are rewritten as pytests.

File: v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/v2g_globals.py [hunk 3]
```diff
@@ -872,28 +874,7 @@ async def __set_select_options(self, entity_id: str, options: list, option_to_se
...
 
-    async def __retrieve_settings(self):
-        """Retrieve all settings from the settings file
-        """
-        self.log(f"__retrieve_settings called")
-
-        if not os.path.exists(self.settings_file_path):
-            self.log(f"__retrieve_settings, no settings file found: creating")
-            self.v2g_settings = {}
-            self.__write_to_file(self.v2g_settings)
```
Comment (rhpijnacker) [inline v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/v2g_globals.py:883]: I left out this line from the refactored implementation: I do not think this is needed.
It only makes sense to write the file if there is actual content to write.

**Review by ArdJonker — APPROVED** [review]

File: v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/settings_manager.py
[the remark this answers is not in the document]
Comment (rhpijnacker) [reply v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/settings_manager.py:1]: Not sure what you mean. There is no "rest".

**Pull request merged**